src/Field/Field.tsx

```diff
@@ -41,9 +41,17 @@ export const Field = ({
       {label && (
         <Box mb={{ custom: outlined ? 4 : 0 }}>
           {renderAsTitle ? (
-            <Text tag="label" typo="heading-small" htmlFor={id} mb="16px">
-              {label}
-            </Text>
+            <Box mb="16px">
+              <Text tag="label" typo="heading-small" htmlFor={id}>
+                {label}
+              </Text>
+
+              {assistiveText && (
```

Comment:
wondering if we should link the two prop from a type point of view?
Something like that if we have `renderAsTitle` then `assistiveText` is optional but available while if `renderAsTitle` is not set we can't even try to set `assistiveText`

Reply:
I've been looking at this this morning and I don't think this is possible given the way our types are constructed. We have interfaces to define the CommonFieldTypes, which are already extending from MarginProps. We'd need to create a discriminated union  like 

```
type AssistiveTextProps = 
| {renderAsTitle: true, assistiveText?: string}
| {renderAsTitle: false, assistiveText?: never}
```

but since the `CommonFieldTypes` is an interface, we can't then extend from a type.

If you have a different idea for how to do it, let me know and I'll tak ea look.

Reply:
If you check the TextInput component we have the exact same scenario?
```
interface Props extends CommonFieldTypes { ... }

type InputProps =
  | {
      /** on change is required and on input optional */
      onChange: (e: string) => void
      onInputChange?: (e: FormEvent<HTMLInputElement>) => void
    }
  | {
      /** on input is required and on change optional */
      onChange?: (e: string) => void
      onInputChange: (e: FormEvent<HTMLInputElement>) => void
    }

export type TextInputProps = Props & InputProps
```

Reply:
Or the problem is that `renderAsTitle` is part of `CommonFieldTypes`?

Reply:
I suppose we could convert everything to types and use type union instead of interfaces? At the end of the day we'll never have to change those types after they are created since it's a library so we can drop interfaces?
@mickaelzhang what do you think?

Reply:
> Or the problem is that `renderAsTitle` is part of `CommonFieldTypes`?

yup, exactly this

Reply:
> 

also, if we do decide to do this, I'll do it as a separate PR.